dockerfiles/prolog/latest/Dockerfile

```diff
@@ -0,0 +1,16 @@
+FROM ubuntu:22.04
+
+# Install necessary packages for SWI-Prolog
+RUN apt-get update \
+    && apt-get -y --no-install-recommends install \
+    build-essential \
+    curl \
+    libgmp-dev \
+    libssl-dev \
+    zlib1g-dev \
```

Comment:
Are these necessary? By installing `swi-prolog`, it should fetch any needed dependencies. But not sure what you plan on running in this image as it is possible you could need depending how you are using it.

Reply:
I noticed that sometimes prolog refused to run without them so I included them and added the --no-recommend flag

Reply:
Do you happen to have any reproducible examples of this? I'm okay with leaving them if they help but just wondering if there was clear examples of issues without it.

Reply:
huh. So I re-ran it and I was mistaken. Prolog works fine without them but the time to build the docker image goes from 22.9s to 44.9s with just
```Dockerfile
RUN apt-get update \
    && apt-get -y install swi-prolog \
    && rm -rf /var/lib/apt/lists/*
```

Reply:
Probably missing `--no-install-recommends`?

Reply:
would that not install needed things tho?

Reply:
That depends what you need. `--no-install-recommends` will still installed all main required dependencies but not optional ones. If you need the optional ones, then you'll have to leave it off or we just leave what code you currently have.

Reply:
no it seems to work fine, I wonder why I added it, maybe I changed something else that fixed it at the same time? In any case I updated the fork (and I also have to wake up in 5 hours so unfortunately I'm also going to bed)